test/browser/features/steps/browser_steps.rb

```diff
@@ -101,3 +101,37 @@
     And the payload has a valid sessions array
   }
 end
+
+Then("the event device ID is valid") do
```

Comment:
This feels bad, but I'm not sure we can do much better. Safari 10 doesn't have local storage available on browserstack, so the device ID is always missing. Selenium can't run javascript via the Ruby API on the mobile browsers, so we can't detect if local storage is available

With browserstack currently we could assume that local storage is available if we're not able to run javascript to check. That would let us assert against device ID on the mobile browsers, but might cause flakes in future if local storage stops being available on them (I might have just been lucky with devices on this PR so far)

Happy to do either — I opted for no flakes but I'm not sure how likely it is to break in future so it might be safe to assume local storage is present instead

Reply:
I think we ought to assume that it can run. At the moment this gives us "false" positives for this feature on mobile browsers without actually testing it.

Reply:
Yeah I think I agree; I'll change it to assume we can use local storage unless we can prove otherwise